proxy/src/transport/tls/config.rs

```diff
@@ -249,15 +249,27 @@ impl CommonConfig {
 
 }
 
+// A Future that, when polled, checks for config updates and publishes them.
+pub type PublishConfigs = Box<Future<Item = (), Error = ()> + Send>;
+
+/// Returns Client and Server config watches, and a task to drive updates.
+///
+/// The returned task Future is expected to never complete.
+///
+/// If there are no TLS settings, then empty watches are returned. In this case, the
+/// Future that is never notified.
+///
+/// If all references are dropped to _either_ the client or server config watches, all
+/// updates will cease for both config watches.
```

Comment:
Ugh...that's not intended. I don't know how much it matters yet.

Reply:
This was my motivation for changing the future semantics such that we should never expect the future to change... i think it's a problem if we're dropping the watch references